drivers/gles3/shaders/scene.glsl

```diff
@@ -1515,6 +1515,14 @@ void main() {
 #CODE : FRAGMENT
 	}
 
+#ifdef DEPTH_USED
+	vec3 ndc = vec3(screen_uv * 2.0 - 1.0, gl_FragDepth);
+	vec4 view_pos = scene_data.inv_projection_matrix * vec4(ndc, 1.0);
+	// Vertex is in view space, the rays are not parallel. It is not enough to update the z component!
+	vertex = view_pos.xyz / view_pos.w;
+	view = -normalize(vertex);
```

Comment:
I'm curious why it is necessary to update `view`. My understanding is everything at that pixel coordinate is some `depth` along the `view` vector. Changing the depth should not affect the view vector.

Your comment notes that "Vertex is in view space, the rays are not parallel. It is not enough to update the z component!"

While you are correct that `vertex` and `view` are in view space, `depth` and `screen_uv` are in *clip space* / screen space, and by changing the z component in *clip space*, we are moving along the `view` ray, so `normalize(view)` should stay constant regardless of how we manipulate `depth` (except perhaps if we push it behind the camera)

Reply:
I'm not sure, I took the original author's word for it.  It seems to work fine without changing view!